File: core-java-practice/leet-code-codebase/LeapYear.java
import java.util.Scanner;

public class LeapYear {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Input year
        System.out.print("Enter a year: ");
        int year = sc.nextInt();

        // Leap year logic
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            System.out.println(year + " is a Leap Year.");
        } else {
            System.out.println(year + " is NOT a Leap Year.");
        }

    }
}

File: core-csharp-practice/gcr-codebase/methods/level02/LeapYear.cs
using System;
class LeapYear{
  public static void Main(string[] args){
    
    //taking input
    Console.Write("Enter a year: ");
    int year = Convert.ToInt32(Console.ReadLine());

    // Check if the year is valid or not
    if (year < 1582){
      Console.WriteLine("Enter other year");
      return;
    }
    bool isLeap = IsLeapYear(year);
    if (isLeap){
      Console.WriteLine(year + " is a Leap Year.");
    }else{
      Console.WriteLine(year + " is not a Leap Year");
    }
  }
  
  // Method to check if a year is a leap year
    static bool IsLeapYear(int y){
    if ((y % 4 == 0 && y % 100 != 0) || (y % 400 == 0)){
      return true;
    }else{
      return false;
    }
  }
}
